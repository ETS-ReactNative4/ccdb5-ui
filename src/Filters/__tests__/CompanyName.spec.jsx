import { shallow } from 'enzyme';
import React from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import ReduxCompanyName, { mapDispatchToProps, CompanyName } from '../CompanyName'

const fixture = [
  { key: "BANK OF", doc_count: 9999 },
  { key: "CAPITAL", doc_count: 999 },
  { key: "Financial", doc_count: 99 },
  { key: "Bankers", doc_count: 9 }
]

function setupEnzyme() {
  const props = {
    forTypeahead: [],
    options: [],
    typeaheadSelect: jest.fn()
  }

  const target = shallow(<CompanyName {...props} />);

  return {
    props,
    target
  }
}

function setupSnapshot() {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: {
      company: ['CAPITAL']
    },
    aggs: {
      company: fixture
    }
  })

  return renderer.create(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ReduxCompanyName />
      </IntlProvider>
    </Provider>
  )
}

describe('component::CompanyName', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const target = setupSnapshot()
      let tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('_onOptionSelected', () => {
    it('checks the filter associated with the option', () => {
      const {target, props} = setupEnzyme()
      target.instance()._onOptionSelected({key: 'foo'})
      expect(props.typeaheadSelect).toHaveBeenCalledWith('foo')
    })
  })

  describe('mapDispatchToProps', () => {
    it('hooks into addMultipleFilters', () => {
      const dispatch = jest.fn()
      mapDispatchToProps(dispatch).typeaheadSelect('foo')
      expect(dispatch.mock.calls.length).toEqual(1)
    })
  })
})
