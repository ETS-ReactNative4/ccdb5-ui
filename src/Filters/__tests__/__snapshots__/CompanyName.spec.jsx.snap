// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::CompanyName snapshots renders without crashing 1`] = `
<section
  className="o-expandable aggregation"
>
  <div
    className="o-expandable_header o-expandable_target"
  >
    <h4
      className="o-expandable_header-left o-expandable_label"
    >
      Matched company name
    </h4>
    <span
      className="o-expandable_header-right o-expandable_link"
    >
      <button
        className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
        onClick={[Function]}
      >
        Hide
        <span
          className="cf-icon cf-icon-minus-round"
        />
      </button>
    </span>
  </div>
  <p>
    The company name as it appears in our complaint system, which may be different than the name the consumer provided in their complaint
  </p>
  <section
    className="typeahead "
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      className="m-btn-inside-input input-contains-label"
    >
      <label
        className="input-contains-label_before input-contains-label_before__search"
      />
      <input
        autoComplete="off"
        className="a-text-input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter company name"
        type="text"
        value=""
      />
    </div>
  </section>
  <ul>
    <li
      className="layout-row m-form-field m-form-field__checkbox"
    >
      <input
        aria-label="CAPITAL"
        checked={true}
        className="flex-fixed a-checkbox"
        id="companyCAPITAL"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="a-label flex-all bucket-key body-copy"
        htmlFor="companyCAPITAL"
      >
        CAPITAL
      </label>
      <span
        className="flex-fixed bucket-count"
      >
        <span>
          999
        </span>
      </span>
    </li>
  </ul>
</section>
`;
